### DISCUSSION TITLE
CreateML model for plank exercises

### DISCUSSION THREAD

**Discussion by christoperBar** [jerrytjahyono/SelfFit#11] [General]
Pada Create Ml terdapat beberapa opsi Model yang dapat digunakan untuk dilatih, pada kasus ini kita perlu melakukan deteksi object apakah sedang melakukan gerakan Plank, yang membingungkan adalah plank itu adalah gerakan yang diam, sedangkan kita perlu hanya mendeteksi plank tersebut tanpa perlu mendeteksi kualitas dari plank itu sendiri seperti apakah plank tersebut adalah plank yang baik, buruk atau lain sebagainya. 

pada kasus ini terjadi kebingungan penggunaan model sebagai berikut :

### Image classification
Penggunaan model melakukan training dataset orang-orang  yang sedang melakukan gerakan Plank berupa gamabar.
Masalahnya kalau pakai model ini nanti labelingnya bagaimana sedangkan hanya mendeteksi 1 gerakan sebagai label, ataukah kemudian kita melakukan dengan 2 label yakni ditambahkan gerakan lainnya yaitu **Leg raises** sebagai label ke-2
### action detection
Penggunaan model melakukan deteksi tindakan yang mungkin terdengar lebih cocok, namun plank adalah gerakan yang berdiam diri ditambah lagi model tersebut masuk kategori Model menggunakan dataset training video.

dari beberapa model yang dapat digunakan manakah yang sebaiknya kita gunakan pada kasus ini ?

**Comment by alizaenazet** [comment]
### Disclaimer
dataset sangat sedikit
<img width="682" alt="Screenshot 2024-05-24 at 16 47 09" src="https://github.com/jerrytjahyono/SelfFit/assets/96227680/18521653-2f1f-45d4-851f-8dd072e54dd1">



https://github.com/jerrytjahyono/SelfFit/assets/96227680/594633f6-3738-4ca1-b86f-b50889e88bdd

_hasil sangat tidak akurat dengan model image classification dengan dataset yang minim_



# Evaluation
<img width="593" alt="Screenshot 2024-05-24 at 16 46 33" src="https://github.com/jerrytjahyono/SelfFit/assets/96227680/09b7598e-e9e6-4eca-b6d4-f87ef51bbc4c">

**Comment by christoperBar** [comment]
ini kalau pakai action classification

https://github.com/jerrytjahyono/SelfFit/assets/146687151/bb0f7e20-6b93-4bee-bfdb-7957a53c1308

hasilnya ada pada video diatas, 
dapat dillihat bahwa hasilnya lebih akurat daripada image classification

apa sudah bener gini ato ada yang lain ya?

**Comment by christoperBar** [comment]
Tes action classification ML exercise sudah lebih akurat dari sebelumnya.

https://github.com/jerrytjahyono/SelfFit/assets/146687151/30c8bd6e-525e-4366-b2ca-9afa43f91683

Namun terdapat ketidak akuratan deteksi pada ML action classification pada gambar dibawah

<img width="1175" alt="Screenshot 2024-05-24 at 23 55 24" src="https://github.com/jerrytjahyono/SelfFit/assets/146687151/93a8b7c3-a73f-47b8-b52c-aa7017b29327">

Untuk label `plank` masih kurang akurat serta pada saat tidak ada objek dalam kamera, akan mendeteksi plank. 
Dataset plank yang kami sediakan adalah video orang plank yang terlihat diam saja.

apakah itu karena video dataset yang kami sediakan pada plank adalah video yang statis(diam)? 
adakah solusi untuk masalah tersebut?

**Comment by christoperBar** [comment]
Saya baru lihat sebuat video dari apple dev WWDC 2021 tentang classify hand pose and action, disitu dijelaskan bahwa ketika pose tangan yang ingin ditraining adalah pose diam(static) maka mereka akan menggunakan hand pose classification. Sedangkan untuk action tangan yang bergerak (seperti dadah👋) mereka akan menggunakan hand action classification.

https://developer.apple.com/videos/play/wwdc2021/10039/

Nah, saya melihat ada kemiripan antara pembahasan mereka dengan problem kita. 

>  Untuk label `plank` masih kurang akurat serta pada saat tidak ada objek dalam kamera, akan mendeteksi plank. 
Dataset plank yang kami sediakan adalah video orang plank yang terlihat diam saja.
apakah itu karena video dataset yang kami sediakan pada plank adalah video yang statis(diam)? 

Apakah saya harus menggunakan model image classification untuk exercise plank? (karena plank adalah exercise yang diam)
Apakah saya harus menggunakan model action classification untuk exercise leg raises? (karena leg raises adalah exercise yang melakukan gerakan kaki)
